### PR TITLE
Proof of concept use contourpy for contouring

### PR DESCRIPTION
**Pull request by dstansby**
See https://github.com/sunpy/sunpy/issues/5874. It didn't take me long to throw this together so thought I'd open a draft PR for discussion. Things we should consider (this is non-exhaustive and off the top of my head; **please add to this**):

- Should we also change `.draw_contour()` to generate the contours through `contourpy`, to make them consistent with the results of `contour()`?

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by Cadair** [comment]
> to make them consistent with the results of contour()?

isn't contourpy the same code mpl uses, so wont it be the same anyway?

**Comment by dstansby** [comment]
> > to make them consistent with the results of contour()?
> 
> isn't contourpy the same code mpl uses, so wont it be the same anyway?

There are three different algorithms:

- MPL to 2014
- MPL 2014 - present
- A new one which is the default and better because it has been written from scratch

I don't think there's any major differences between them, but if we use the default new algorithm for `.contour()` and the `Axes.contour()` default for drawing contours we might end up with slightly different contours.

**Comment by dstansby** [comment]
For reference, here's the Matplotlib issue and pull request:

https://github.com/matplotlib/matplotlib/issues/22529
https://github.com/matplotlib/matplotlib/pull/22567

**Comment by Cadair** [comment]
Happy to change how we do it for now and drop back to the new mpl behaviour in ~18 months when we can depend on it in mpl?

File: setup.cfg [hunk 1]
```diff
@@ -55,6 +55,7 @@ image =
 jpeg2000 =
   glymur>=0.8.18,!=0.9.0,!=0.9.5
 map =
+  contourpy>=1
```
Comment (ianthomas23) [inline setup.cfg:58]: This should now be `contourpy>=1.0.1`. I had an unnecessary `numpy` version pin in `contourpy` which might cause problems, the new release fixes this.

**Comment by ianthomas23** [comment]
Contourpy dev here. Do you folks want mpl's `contour/contourf` to have a new kwarg for the contourpy algorithm name? I have been assuming that mpl will switch to using the new algorithm asap and that a kwarg isn't needed as it means explaining the algorithm options to end users who mostly just want it to work. But if there is a use case for the kwarg being there, I am happy to change my mind.

**Comment by Cadair** [comment]
:wave: 

If mpl switches to the new algorithm then I think it has the same effect for us, when we depend on that version of mpl we can say they do the same thing, so I think that approach is fine with us :)

**Comment by nabobalis** [comment]
If we plan to work around the main issue that we have with scikit-image, do we just wait for MPL to depend on contourpy and we close this PR?

**Comment by dstansby** [comment]
I'm not bothered whether this makes it into 4.0; the only good reason to rush seems to be removing the `scikit-image` dependency, which I don't think will happen for 4.0, so I'm going to re-milestone this but leave it open.

**Comment by nabobalis** [comment]
I am happy to get this into 4.0 if we decide that removing scikit-image is important enough, if not, waiting for MPL to do the switch is just easier for us? (We would still need to update the contour function tho)

**Comment by dstansby** [comment]
I think we should hold off on this until MPL makes the switch that won't be before we feature freeze 4.0, so I'm going to close this PR and we can track this at https://github.com/sunpy/sunpy/issues/5874

**Pull request closed without merging**